tests/foreman/ui/test_organization.py

```diff
@@ -376,41 +432,133 @@ def test_negative_update(self):
 
     @run_only_on('sat')
     @tier2
-    def test_positive_remove_domain(self):
+    def test_positive_create_with_subnet(self):
+        """Create Org with subnet.
+
+        :id: 6736cd82-a2b0-4fc0-a2bc-99c9f13464d7
+
+        :expectedresults: Org created with subnet.
+
+        :CaseLevel: Integration
+        """
+        with Session(self.browser) as session:
+            for subnet_name in generate_strings_list():
+                with self.subTest(subnet_name):
+                    # Create subnet using nailgun
+                    subnet = entities.Subnet(
+                        name=subnet_name,
+                        network=gen_ipaddr(ip3=True),
+                        mask='255.255.255.0'
+                    ).create()
+                    self.assert_org_creation_with(
+                        session, subnet, subnet_name, 'context.tab_subnets',
+                        subnets=[subnet_name]
+                    )
+
+    @run_only_on('sat')
+    @tier2
+    def test_positive_update_subnet(self):
+        """Add and Remove subnet by using organization name and subnet name.
+
+        :id: bc59bdeb-b538-4473-a096-e4de2454497d
+
+        :expectedresults: subnet is added then removed.
+
+        :CaseLevel: Integration
+        """
+        with Session(self.browser) as session:
+            org_name = gen_string('alpha')
+            make_org(session, org_name=org_name)
+            for subnet_name in generate_strings_list():
+
+                with self.subTest(subnet_name):
+                    # Create subnet using nailgun
+                    subnet = entities.Subnet(
+                        name=subnet_name,
+                        network=gen_ipaddr(ip3=True),
+                        mask='255.255.255.0',
+                    ).create()
+                    self.assert_org_update_with(
+                        org_name, session, subnet, subnet_name,
+                        'context.tab_subnets', subnets=[subnet_name]
+                    )
```

Comment:
Personally i'm not too happy about such assertions. We're always trying to keep tests as much simple as possible and such custom assertions aren't clear at all - without looking through their implementation i can't say what's going on in this test.
Just take a look - it's called `assert_org_update_with` but according to docstring test is not updating but removing the subnet, those args are not user-friendly too - helper accepts everything starting from session and ending with string containing key to locator of some tab.
I've recently had a discussion with @rplevka and his point was to avoid introducing such magic helpers to keep test steps clear without need to analyze helpers source code to understand them. Summoning @rplevka here too. :) Roman, what do you think about this case?